src/lib/src/service/persistence/persistence.service.ts

```diff
@@ -29,12 +37,31 @@ export class PersistenceService {
     return this.getHttpClient().get<WorkspaceElement>(this.listFilesUrl).toPromise();
   }
 
-  createResource(path: string, type: string): Promise<string> {
-    return this.getHttpClient().post(this.getURL(path), '', { responseType: 'text', params: { type: type } }).toPromise();
+  createResource(path: string, type: string): Observable<string | Conflict> {
+    return this.getHttpClient().post(this.getURL(path), '', {
+      observe: 'response',
+      responseType: 'text',
+      params: { type: type }
+    }).map(response => response.body).catch(response => {
```

Comment:
I'd prefer to have .catch in a new line (in order to separate it from the mapping).